src/lib/updateReasonNames.test.js

```diff
@@ -0,0 +1,160 @@
+const mockUser = {
+  id: 1000,
+  homeRegionId: 1,
+  name: 'chud',
+  hsesUsername: 'chud',
+  hsesUserId: '1000',
+};
+
+const reportObject = {
+  regionId: 1,
+  activityRecipientType: 'grantee',
+  status: REPORT_STATUSES.APPROVED,
+  userId: mockUser.id,
+  lastUpdatedById: mockUser.id,
+  ECLKCResourcesUsed: ['test'],
+  activityRecipients: [
+    { activityRecipientId: GRANTEE_ID },
+    { activityRecipientId: GRANTEE_ID_TWO },
+  ],
+  approvingManagerId: 1,
+  numberOfParticipants: 11,
+  deliveryMethod: 'method',
+  duration: 1,
+  endDate: '2000-01-01T12:00:00Z',
+  startDate: '2000-01-01T12:00:00Z',
+  requester: 'requester',
+  programTypes: ['type'],
+  targetPopulations: ['pop'],
+  reason: ['reason'],
+  participants: ['participants'],
+  topics: ['topics'],
+  ttaType: ['technical-assistance'],
+};
+
+describe('update reason names job', () => {
+  beforeAll(async () => {
+    await User.findOrCreate({ where: mockUser });
+    await Grantee.findOrCreate({ where: { name: 'grantee', id: GRANTEE_ID } });
+
+    await Grant.findOrCreate({
+      where: {
+        id: GRANTEE_ID, number: '1', granteeId: GRANTEE_ID, regionId: 1, status: 'Active',
+      },
+    });
+    await Grant.findOrCreate({
+      where: {
+        id: GRANTEE_ID_TWO, number: '2', granteeId: GRANTEE_ID, regionId: 1, status: 'Active',
+      },
+    });
+  });
+
+  afterAll(async () => {
+    const reports = await ActivityReport
+      .findAll({ where: { userId: [mockUser.id] } });
+    const ids = reports.map((report) => report.id);
+    await NextStep.destroy({ where: { activityReportId: ids } });
+    await ActivityRecipient.destroy({ where: { activityReportId: ids } });
+    await ActivityReport.destroy({ where: { id: ids } });
+    await User.destroy({ where: { id: [mockUser.id] } });
+    await Grant.destroy({ where: { id: [GRANTEE_ID, GRANTEE_ID_TWO] } });
+    await Grantee.destroy({ where: { id: [GRANTEE_ID, GRANTEE_ID_TWO] } });
+    await db.sequelize.close();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('updates legacy reasons in the database', async () => {
+    // Req 1- Replace All.
+    await ActivityReport.create({
+      ...reportObject,
+      requester: 'Req 1',
+
+      reason: [
+        'Monitoring | Under Enrollment',
+        'Data Training or Analysis',
+        'Professional Development',
+        'Quality Improvement Support',
+        'Program Planning',
+        'School Readiness',
+      ],
+    });
+
+    // Req 2 - Mixed bag keep DISTINCT Non Legacy Reasons.
+    await ActivityReport.create({
+      ...reportObject,
+      requester: 'Req 2',
+
+      reason: [
+        'Child Incidents',
+        'Full Enrollment',
+        'Monitoring | Under Enrollment',
+        'Data Training or Analysis',
+        'Quality Improvement Support',
+        'Ongoing Quality Improvement',
+        'New Program Option',
+      ],
+    });
+
+    // Req 3 - Only new Reasons (no change).
+    await ActivityReport.create({
```

Comment:
I don't see any assertions against this ActivityReport (or the next one). I'd feel more comfortable with the change if we had an assertion that reports with no legacy reasons are not updated.

Reply:
I have updated the unit test. Initially was going against the count as it only returns AR's changed.